sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepository.kt

```diff
@@ -125,10 +125,15 @@ internal class FileCompactBlockRepository(
          */
         const val BLOCK_FILENAME_SUFFIX = "-compactblock"
 
+        // TODO [#1555]: Keep BLOCKS_METADATA_BUFFER_SIZE the same as current SYNC_BATCH_SIZE
+        // TODO [#1555]: https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/issues/1555
+
         /**
          * The size of block meta data buffer
+         *
+         * Note: we should keep this size as close as possible to [CompactBlockProcessor.SYNC_BATCH_SIZE]
          */
-        const val BLOCKS_METADATA_BUFFER_SIZE = 10
+        const val BLOCKS_METADATA_BUFFER_SIZE = 1000
```

Comment:
I was concerned that this might cause problems during sandblasting (note that both tested ranges start after sandblasting), because #1555 is not handled, and we aren't actually scanning all sandblasting blocks. The change to `BLOCKS_METADATA_BUFFER_SIZE` means that this code will never trigger for the 100-block sandblasting ranges:

https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/blob/f4086a179c4724cb85a64909b32297b323aefbff/sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepository.kt#L46-L49
https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/blob/f4086a179c4724cb85a64909b32297b323aefbff/sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepository.kt#L161-L163

However, it turns out that we are fine, because there is a catch-all that writes out whatever is left in the metadata buffer:

https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/blob/f4086a179c4724cb85a64909b32297b323aefbff/sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepository.kt#L52-L55

What this means is that whenever `BLOCKS_METADATA_BUFFER_SIZE` is greater than or equal to the batch size being written, we always have the desired behaviour: write all blocks to disk, then flush the metadata to the Rust side in a single go.

Reply:
Yes, you read this correctly. We're safe in changing BLOCKS_METADATA_BUFFER_SIZE size.